From 27ce7a64ef6c6bd74e311a396298de90aa696846 Mon Sep 17 00:00:00 2001
From: Carl Steinbach <cws@apache.org>
Date: Mon, 11 Jun 2012 20:32:40 +0000
Subject: [PATCH 127/144] HIVE-3013 [jira] TestCliDriver cannot be debugged with eclipse since hadoop_home is set wrongly

Summary:
HIVE-3013

fix typo

cp Fix

cp

Test Plan: EMPTY

Reviewers: JIRA, njain, kevinwilfong

Differential Revision: https://reviews.facebook.net/D3555

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1348995 13f79535-47bb-0310-9956-ffa450edef68
---
 build.properties                                   |    6 ++
 build.xml                                          |   11 ++++
 eclipse-templates/.classpath                       |    2 +
 eclipse-templates/HiveCLI.launchtemplate           |    6 ++-
 eclipse-templates/HiveServer.launchtemplate        |   50 ++++++++++++++++++++
 eclipse-templates/TestCliDriver.launchtemplate     |    5 +-
 .../TestEmbeddedHiveMetaStore.launchtemplate       |    5 +-
 .../TestHBaseCliDriver.launchtemplate              |    5 +-
 eclipse-templates/TestHive.launchtemplate          |    5 +-
 .../TestHiveMetaStoreChecker.launchtemplate        |    5 +-
 eclipse-templates/TestJdbc.launchtemplate          |    5 +-
 eclipse-templates/TestMTQueries.launchtemplate     |    5 +-
 .../TestRemoteHiveMetaStore.launchtemplate         |    5 +-
 eclipse-templates/TestTruncate.launchtemplate      |    5 +-
 14 files changed, 100 insertions(+), 20 deletions(-)
 create mode 100644 eclipse-templates/HiveServer.launchtemplate

diff --git a/src/build.properties b/src/build.properties
index 08dd641..98f85b7 100644
--- a/src/build.properties
+++ b/src/build.properties
@@ -105,3 +105,9 @@ mvn.license.dir=${build.dir.hive}/maven/licenses
 #
 datanucleus.repo=http://www.datanucleus.org/downloads/maven2
 
+#
+# Eclipse Properties
+#
+
+# JVM arguments for Eclipse launch configurations
+eclipse.launch.jvm.args=-Xms256m -Xmx1024m
diff --git a/src/build.xml b/src/build.xml
index d9ec1b5..464a545 100644
--- a/src/build.xml
+++ b/src/build.xml
@@ -606,6 +606,14 @@
       <matches string="${hadoop.version.ant-internal}" pattern="^0\.20\..*" />
     </condition>
 
+
+    <!-- Construct the classpath needed for testutils/hadoop launch script -->
+    <path id="hive.hadoop.test.classpath">
+      <fileset dir="${hive.root}/build/ivy/lib/hadoop0.${hadoop.mr.rev}.shim" includes="*.jar" />
+      <fileset dir="${hive.root}/build/ivy/lib/default" includes="*.jar" excludes="hive*.jar" />
+    </path>
+    <property name="hive.hadoop.test.classpath" refid="hive.hadoop.test.classpath"/>
+    
     <pathconvert property="eclipse.project">
       <path path="${basedir}"/>
       <regexpmapper from="^.*/([^/]+)$$" to="\1" handledirsep="yes"/>
@@ -624,6 +632,9 @@
         <filter token="JETTYUTILJAR" value="${jetty.util.jar}"/>
         <filter token="SERVLETAPIJAR" value="${servlet-api.jar}"/>
         <filter token="HIVE_VERSION" value="${version}"/>
+        <filter token="HIVE_HADOOP_TEST_CLASSPATH" value="${hive.hadoop.test.classpath}"/>
+        <filter token="HADOOP_BIN_PATH" value="${hive.root}/testutils/hadoop"/>
+        <filter token="JVM_ARGS" value="${eclipse.launch.jvm.args}"/>
       </filterset>
     </copy>
     <move todir="." includeemptydirs="false">
diff --git a/src/eclipse-templates/.classpath b/src/eclipse-templates/.classpath
index 3f7ae42..ce9de61 100644
--- a/src/eclipse-templates/.classpath
+++ b/src/eclipse-templates/.classpath
@@ -46,6 +46,8 @@
   <classpathentry kind="lib" path="build/ivy/lib/hadoop0.20.shim/oro-2.0.8.jar"/>
   <classpathentry kind="lib" path="build/ivy/lib/hadoop0.20.shim/servlet-api-2.5-6.1.14.jar"/>
   <classpathentry kind="lib" path="build/ivy/lib/hadoop0.20.shim/xmlenc-0.52.jar"/>
+  <classpathentry kind="lib" path="build/ivy/lib/default/avro-@avro.version@.jar"/>
+  <classpathentry kind="lib" path="build/ivy/lib/default/avro-mapred-@avro.version@.jar"/>
   <classpathentry kind="lib" path="build/ivy/lib/default/jline-@jline.version@.jar"/>
   <classpathentry kind="lib" path="build/ivy/lib/default/sqlline-@sqlline.version@.jar"/>
   <classpathentry kind="lib" path="build/ivy/lib/default/json-@json.version@.jar"/>
diff --git a/src/eclipse-templates/HiveCLI.launchtemplate b/src/eclipse-templates/HiveCLI.launchtemplate
index 05cb373..507a225 100644
--- a/src/eclipse-templates/HiveCLI.launchtemplate
+++ b/src/eclipse-templates/HiveCLI.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
 
   <listAttribute key="org.eclipse.debug.core.MAPPED_RESOURCE_PATHS">
@@ -35,6 +35,7 @@
     <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry containerPath=&quot;org.eclipse.jdt.launching.JRE_CONTAINER&quot; javaProject=&quot;@PROJECT@&quot; path=&quot;1&quot; type=&quot;4&quot;/&gt;&#10;"/>
     <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/conf&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
     <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/common/src/java/conf&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
+    <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/build/metastore/hive-metastore-@HIVE_VERSION@.jar&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
     <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry id=&quot;org.eclipse.jdt.launching.classpathentry.defaultClasspath&quot;&gt;&#10;&lt;memento exportedEntriesOnly=&quot;false&quot; project=&quot;@PROJECT@&quot;/&gt;&#10;&lt;/runtimeClasspathEntry&gt;&#10;"/>
     <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/build/ql/hive-exec-@HIVE_VERSION@.jar&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
   </listAttribute>
@@ -43,6 +44,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.cli.CliDriver"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+    value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@"/>
 
 </launchConfiguration>
diff --git a/src/eclipse-templates/HiveServer.launchtemplate b/src/eclipse-templates/HiveServer.launchtemplate
new file mode 100644
index 0000000..a550c6f
--- /dev/null
+++ b/src/eclipse-templates/HiveServer.launchtemplate
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+ 
+      http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--> 
+<launchConfiguration type="org.eclipse.jdt.launching.localJavaApplication">
+  <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
+  <mapAttribute key="org.eclipse.debug.core.environmentVariables">
+    <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
+  </mapAttribute>
+
+  <listAttribute key="org.eclipse.debug.core.MAPPED_RESOURCE_PATHS">
+    <listEntry value="/hive/service/src/java/org/apache/hadoop/hive/service/HiveServer.java"/>
+  </listAttribute>
+
+  <listAttribute key="org.eclipse.debug.core.MAPPED_RESOURCE_TYPES">
+    <listEntry value="1"/>
+  </listAttribute>
+
+    <listAttribute key="org.eclipse.jdt.launching.CLASSPATH">
+    <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry containerPath=&quot;org.eclipse.jdt.launching.JRE_CONTAINER&quot; javaProject=&quot;@PROJECT@&quot; path=&quot;1&quot; type=&quot;4&quot;/&gt;&#10;"/>
+    <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/conf&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
+    <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/common/src/java/conf&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
+    <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/build/metastore/hive-metastore-@HIVE_VERSION@.jar&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
+    <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry id=&quot;org.eclipse.jdt.launching.classpathentry.defaultClasspath&quot;&gt;&#10;&lt;memento exportedEntriesOnly=&quot;false&quot; project=&quot;@PROJECT@&quot;/&gt;&#10;&lt;/runtimeClasspathEntry&gt;&#10;"/>
+    <listEntry value="&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#10;&lt;runtimeClasspathEntry internalArchive=&quot;/@PROJECT@/build/ql/hive-exec-@HIVE_VERSION@.jar&quot; path=&quot;3&quot; type=&quot;2&quot;/&gt;&#10;"/>
+  </listAttribute>
+
+
+  <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
+  <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.service.HiveServer"/>
+  <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+    value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@"/>
+
+</launchConfiguration>
diff --git a/src/eclipse-templates/TestCliDriver.launchtemplate b/src/eclipse-templates/TestCliDriver.launchtemplate
index 83bd0ef..2db3456 100644
--- a/src/eclipse-templates/TestCliDriver.launchtemplate
+++ b/src/eclipse-templates/TestCliDriver.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.cli.TestCliDriver"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
diff --git a/src/eclipse-templates/TestEmbeddedHiveMetaStore.launchtemplate b/src/eclipse-templates/TestEmbeddedHiveMetaStore.launchtemplate
index 65b6a50..3dc1010 100644
--- a/src/eclipse-templates/TestEmbeddedHiveMetaStore.launchtemplate
+++ b/src/eclipse-templates/TestEmbeddedHiveMetaStore.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.metastore.TestEmbeddedHiveMetaStore"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
diff --git a/src/eclipse-templates/TestHBaseCliDriver.launchtemplate b/src/eclipse-templates/TestHBaseCliDriver.launchtemplate
index 586922f..ab02c4d 100644
--- a/src/eclipse-templates/TestHBaseCliDriver.launchtemplate
+++ b/src/eclipse-templates/TestHBaseCliDriver.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.cli.TestHBaseCliDriver"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
diff --git a/src/eclipse-templates/TestHive.launchtemplate b/src/eclipse-templates/TestHive.launchtemplate
index dc91f4a..5e68c21 100644
--- a/src/eclipse-templates/TestHive.launchtemplate
+++ b/src/eclipse-templates/TestHive.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.ql.metadata.TestHive"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
diff --git a/src/eclipse-templates/TestHiveMetaStoreChecker.launchtemplate b/src/eclipse-templates/TestHiveMetaStoreChecker.launchtemplate
index af94271..2bb9268 100644
--- a/src/eclipse-templates/TestHiveMetaStoreChecker.launchtemplate
+++ b/src/eclipse-templates/TestHiveMetaStoreChecker.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.ql.metadata.TestHiveMetaStoreChecker"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
diff --git a/src/eclipse-templates/TestJdbc.launchtemplate b/src/eclipse-templates/TestJdbc.launchtemplate
index 5331b89..267f58a 100644
--- a/src/eclipse-templates/TestJdbc.launchtemplate
+++ b/src/eclipse-templates/TestJdbc.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,7 +37,8 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.jdbc.TestJdbcDriver"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot; -Dtest.warehouse.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/test/data/warehouse&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot; -Dtest.warehouse.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/test/data/warehouse&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
 
diff --git a/src/eclipse-templates/TestMTQueries.launchtemplate b/src/eclipse-templates/TestMTQueries.launchtemplate
index 81e5b18..172692e 100644
--- a/src/eclipse-templates/TestMTQueries.launchtemplate
+++ b/src/eclipse-templates/TestMTQueries.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.ql.TestMTQueries"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
diff --git a/src/eclipse-templates/TestRemoteHiveMetaStore.launchtemplate b/src/eclipse-templates/TestRemoteHiveMetaStore.launchtemplate
index 5f06d03..81e3877 100644
--- a/src/eclipse-templates/TestRemoteHiveMetaStore.launchtemplate
+++ b/src/eclipse-templates/TestRemoteHiveMetaStore.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.metastore.TestRemoteHiveMetaStore"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dtest.warehouse.dir=&quot;pfile://${workspace_loc:@PROJECT@}/build/test/data/warehouse&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
diff --git a/src/eclipse-templates/TestTruncate.launchtemplate b/src/eclipse-templates/TestTruncate.launchtemplate
index f6b8d09..38513a5 100644
--- a/src/eclipse-templates/TestTruncate.launchtemplate
+++ b/src/eclipse-templates/TestTruncate.launchtemplate
@@ -20,7 +20,7 @@
   <booleanAttribute key="org.eclipse.debug.core.appendEnvironmentVariables" value="false"/>
   <mapAttribute key="org.eclipse.debug.core.environmentVariables">
     <mapEntry key="JAVA_HOME" value="${system_property:java.home}"/>
-    <mapEntry key="HADOOP_HOME" value="${workspace_loc:@PROJECT@}/build/ivy/lib/hadoop.@HADOOPVER@.shim"/>
+    <mapEntry key="HIVE_HADOOP_TEST_CLASSPATH" value="@HIVE_HADOOP_TEST_CLASSPATH@"/>
   </mapAttribute>
   <stringAttribute key="org.eclipse.jdt.junit.CONTAINER" value=""/>
   <booleanAttribute key="org.eclipse.jdt.junit.KEEPRUNNING_ATTR" value="false"/>
@@ -37,6 +37,7 @@
   <booleanAttribute key="org.eclipse.jdt.launching.DEFAULT_CLASSPATH" value="false"/>
   <stringAttribute key="org.eclipse.jdt.launching.MAIN_TYPE" value="org.apache.hadoop.hive.metastore.TestTruncate"/>
   <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="@PROJECT@"/>
-  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Dhive.root.logger=INFO,console -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
+  <stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS"
+                   value="@JVM_ARGS@ -Dhive.root.logger=INFO,console -Dhadoop.bin.path=@HADOOP_BIN_PATH@ -Dtest.tmp.dir=&quot;${workspace_loc:@PROJECT@}/build/ql/tmp&quot; -Dbuild.dir=&quot;${workspace_loc:@PROJECT@}/build/ql&quot; -Dbuild.dir.hive=&quot;${workspace_loc:@PROJECT@}/build&quot; -Dversion=&quot;@HIVE_VERSION@&quot;"/>
   <stringAttribute key="org.eclipse.jdt.launching.WORKING_DIRECTORY" value="${workspace_loc:@PROJECT@}/ql"/>
 </launchConfiguration>
-- 
1.7.0.4

